### DISCUSSION TITLE
Can livewire emit() then redirect()->route('path')

### DISCUSSION THREAD

**Discussion by jayenne** [livewire/livewire#2137] [1. Help] [answered]
Hi all,

I have a working component (alert/toast) that performs a task, then emit() a response then return back() and all is well.

but when I return redirect()->route('path') I don't seems to be getting the emit() - should I? or should I doing something else to pass data back to a different route?

fyi: my component is included on every page (via app.blade) the trigger is on sleected page and the redirect does not include the originating trigger.

**Comment by joshhanley** [comment, marked as answer]
@jayenne you're better off using the session to flash the data in that instance as emit only works while Livewire components are running, but a redirect will cause a full page refresh.

Hope this helps!

**Reply by jayenne** [reply]
yup, that helps thanks.
